### ISSUE TITLE
How to mirror about X axis?

### ISSUE THREAD

**Issue by dj-kefir-siorbacz** [Wasserwecken/bvhio#6]
Hi @Wasserwecken ! 
Qucik a question - how to mirror the whole animation around X axis? 
I would think that scaling the `Restpose` with (-1,1,1) would do the job, but If I'm not mistaken I can only assign a scalar value to `Scale` property.

**Comment by Wasserwecken** [comment]
Yes, you can set either a Tuple for the scale OR a scalar. `joint.Scale = (-1, 1, 1)` should definitly work.

I had the same issue of mirroring animation data for my master thesis. Its not that straight forward, and I did with pytorch tensors. BUT in theory you should be right by setting the root rest pose scale to (-1, 1, 1). You might have to invert the root position animation data too, I am not definitly sure.

Please note that if the rest pose has a negative scale, all the other data will be untouched! If you want to process later data that required the local space of the joints, they are not mirrored! If you want to convert the data completly you may want to use following line: `root.applyRestposeScale(recursive=True, bake=False, bakeKeyframes=True)`

Heres my code for mirroring with pytorch, but that was outside of bvhio with my my own internal tensor representation (may it helps, but probably not):
```py
def _mirror(self, spaces: torch.Tensor):
        leftIndicies = [key for key, value in self.JointNames.items() if self.LeftRightJoints[0] in value]
        rightIndicies = [key for key, value in self.JointNames.items() if self.LeftRightJoints[1] in value]

        mirror = spaces.detach().clone()
        mirmat = torch.tensor(((  1, -1, -1, 0),
                               ( -1,  1,  1, 0),
                               ( -1,  1,  1, 0),
                               ( -1,  1,  1, 1),))

        for left, right in zip(leftIndicies, rightIndicies):
            mirror[:-1, ..., left, :, :] = spaces[:-1, ..., right, :, :]
            mirror[:-1, ..., right, :, :] = spaces[:-1, ..., left, :, :]
        mirror[:-1, ..., :, :, :] *= mirmat

        return mirror
```

**Comment by dj-kefir-siorbacz** [comment]
Hi @Wasserwecken,

I think when I do:
```
root.RestPose.Scale = (-1,-1,1)
root.applyRestposeScale(recursive=True, bakeKeyframes=False, bake=False)
```
The Yposition channel of Hips is modified in the MOTION section. No idea why?

**Comment by dj-kefir-siorbacz** [comment]
This solves my above issue, but is not elegant: 
```python
from copy import deepcopy

kf = deepcopy(root.Keyframes)
root.RestPose.Scale = (-1,-1,1)
root.applyRestposeScale(recursive=True, bakeKeyframes=False, bake=False)
root.Keyframes = kf
```

**Comment by Wasserwecken** [comment]
If you change the scale of a joint / transform the position will be untouched, but all the child position data will be scaled because they will depend on that.

``` py
    def applyScale(self, scale: glm.vec3 = None, recursive: bool = False, bake: bool = False) -> "Transform":
        """Changes the scale of the transform and updates its children to keep them spatial unchanged.
        - If scale is NOT set -> The transform resets its scale to (1, 1, 1).
        - If scale IS set -> The given scale is added to the current scale.
        - If bake Is True -> The scale correction is NOT passed to the children, only positions will be modified.

        Returns itself."""
        change, changeInverse = self._applyScaleGetChanges(scale)

        self._applyScaleChange(change)

        for child in self.Children:
            child._applyScaleChangeInverse(changeInverse, bake=bake)

            if recursive:
                child.applyScale(scale=scale, recursive=True, bake=bake)

        return self
```

That said, if your "root"-joint is not your "hip"-joint, then the scale on the root will be passed to the hips, because the hips are related to the root. But you probably want that! Because if you scale your selekton 10 times down and the hips are staying in the same position, your feets will floating in the air!

Another important note of mirroring the skeleton!! IF you apply the scale of (-1, 1, 1) and it "works" remember that the "left arm" is still the "left arm"! When you process this data further remember to also switch the left and right labeling! I've done this too, just refer to the code of my first answer of this issue.
